util/include/cppmicroservices/util/BundleElfFile.h

```diff
@@ -222,21 +228,33 @@ class BundleElfFile
       return iter->second;
     }
 
-    Shdr* strTblHdr = m_SectionHeaders + strTblHdrIdx;
+    const Shdr* const strTblHdr = m_SectionHeaders.get() + strTblHdrIdx;
+
+    if (strTblHdr && 0 > strTblHdr->sh_size) {
+      return nullptr;
+    }
 
     fs.seekg(strTblHdr->sh_offset);
-    char* strTbl = new char[static_cast<std::size_t>(strTblHdr->sh_size)];
-    fs.read(strTbl, strTblHdr->sh_size);
-    m_StrTblIndexToStrArray.insert(std::make_pair(strTblHdrIdx, strTbl));
+    char* strTbl = nullptr;
+    try {
+        strTbl = new char[static_cast<std::size_t>(strTblHdr->sh_size)];
+        fs.read(strTbl, strTblHdr->sh_size);
+        m_StrTblIndexToStrArray.insert(std::make_pair(strTblHdrIdx, strTbl));
+    } catch (...) {
+      if (strTbl) {
+        delete[] strTbl;
```

Comment:
You can get rid of `try..catch..` if `strTbl` is a unique_ptr to the new array? And return the unique_ptr from the function and make corresponding adjustments at the call-site (looks like there is only one)

Reply:
I can't do that since ```m_StrTblIndexToStrArray``` owns ```strTbl```.  Since there is only one call to ```GetStringTable```, I can do away with ```m_StrTblIndexToStrArray``` altogether and return ```std::unique_ptr<char[]>```. There isn't a need to cache the string table right now anyway (which I believe was the point of ```m_StrTblIndexToStrArray```).